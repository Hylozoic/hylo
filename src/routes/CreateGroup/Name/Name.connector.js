import { connect } from 'react-redux'
import { push } from 'connected-react-router'
import { addGroupName, addParentIds } from '../CreateGroup.store'
import getRouteParam from 'store/selectors/getRouteParam'
import { get } from 'lodash/fp'

export function mapStateToProps (state, props) {
  const parentId = getRouteParam('groupId', state, props)
  return {
    parentIds: [parentId],
    groupName: get('name', state.CreateGroup)
  }
}

export function mapDispatchToProps (dispatch, props) {
  return {
    goToNextStep: () => dispatch(push('/create-group/domain')),
    goHome: () => dispatch(push('/')),
    addGroupName: name => dispatch(addGroupName(name)),
    addParentIds: parentIds => dispatch(addParentIds(parentIds))
  }
}

export function mergeProps (stateProps, dispatchProps, ownProps) {
  const { parentId } = stateProps
  const addParentIds = parentId
    ? () => dispatchProps.addParentIds([parentId])
    : () => {}

  return {
    ...stateProps,
    ...dispatchProps,
    ...ownProps,
    addParentIds
  }
}

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)
