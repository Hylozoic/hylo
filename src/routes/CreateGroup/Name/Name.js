import React, { Component } from 'react'
import '../CreateGroup.scss'
import TextInput from 'components/TextInput'
import { hyloNameWhiteBackground, happyAxolotl } from 'util/assets'
import { bgImageStyle } from 'util/index'
import ModalFooter from 'components/ModalFooter'
import ModalSidebar from 'components/ModalSidebar'

export default class Name extends Component {
  constructor (props) {
    super(props)
    this.state = {
      groupName: ''
    }
  }

  handleNameChange = (event) => {
    const groupName = event.target.value
    this.setState({
      groupName
    })
  }

  submit = () => {
    this.props.addGroupName(this.state.groupName)
    this.props.goToNextStep()
  }

  errorCheckAndSubmit = () => {
    if (this.state.groupName === '') {
      this.setState({
        error: 'Please enter a group name'
      })
    } else {
      this.submit()
    }
  }

  onEnter = event => {
    if (event.key === 'Enter') {
      this.errorCheckAndSubmit()
    }
  }

  componentWillMount = () => {
    const { groupName } = this.props
    if (groupName) this.setState({ groupName })
  }

  componentDidMount = () => {
    this.props.addParentIds()
  }

  render () {
    return <div styleName='flex-wrapper'>
      <ModalSidebar
        imageDialogOne={"Hi, I'm Hylo the Axolotl!"}
        imageDialogTwo={"It's great to meet you!"}
        imageUrl={happyAxolotl}
        onClick={this.props.goHome}
        theme={sidebarTheme}
        header="Great, let's get started"
        body="All good things start somewhere! Let's kick things off with a catchy name for your group."
      />
      <div styleName='panel'>
        <div>
          <span styleName='step-count'>STEP 1/3</span>
        </div>
        <div styleName='center'>
          <div styleName='logo center' style={bgImageStyle(hyloNameWhiteBackground)} />
        </div>
        <div styleName='center-vertically'>
          <span styleName='text-input-label'>{ this.state.groupName && "What's the name of your group?"}</span>
          <TextInput
            type='text'
            name='group-name'
            onChange={this.handleNameChange}
            value={this.state.groupName}
            theme={inputTheme}
            placeholder="What's the name of your group?"
            noClearButton
            onEnter={this.onEnter}
          />
          { this.state.error && <span styleName='arrow-up' /> }
          { this.state.error && <span styleName='error'>{this.state.error}</span>}
        </div>
      </div>
      <ModalFooter
        submit={this.errorCheckAndSubmit}
        showPrevious={false}
        continueText={'Continue'}
      />
    </div>
  }
}

const inputTheme = {
  inputStyle: 'modal-input',
  wrapperStyle: 'center'
}

const sidebarTheme = {
  sidebarHeader: 'sidebar-header-full-page',
  sidebarText: 'gray-text sidebar-text-full-page'
}
