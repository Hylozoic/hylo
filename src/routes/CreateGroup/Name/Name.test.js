import React from 'react'
import { shallow } from 'enzyme'
import Name from './Name'

describe('Name', () => {
  const groupName = 'groupName'
  it('renders correctly', () => {
    const wrapper = shallow(<Name
      groupName={groupName}
      goToNextStep={jest.fn()}
      addGroupName={jest.fn()}
      goHome={jest.fn()}
      addNetworkId={() => {}}
    />)
    expect(wrapper).toMatchSnapshot()
  })
})
