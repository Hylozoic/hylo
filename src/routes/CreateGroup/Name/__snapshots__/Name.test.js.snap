// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name renders correctly 1`] = `
<div
  data-stylename="flex-wrapper"
>
  <ModalSidebar
    body="All good things start somewhere! Let's kick things off with a catchy name for your group."
    header="Great, let's get started"
    imageDialogOne="Hi, I'm Hylo the Axolotl!"
    imageDialogTwo="It's great to meet you!"
    imageUrl="/happy-axolotl.png"
    onClick={[MockFunction]}
    theme={
      Object {
        "sidebarHeader": "sidebar-header-full-page",
        "sidebarText": "gray-text sidebar-text-full-page",
      }
    }
  />
  <div
    data-stylename="panel"
  >
    <div>
      <span
        data-stylename="step-count"
      >
        STEP 1/3
      </span>
    </div>
    <div
      data-stylename="center"
    >
      <div
        data-stylename="logo center"
        style={
          Object {
            "backgroundImage": "url(/logo-white.svg)",
          }
        }
      />
    </div>
    <div
      data-stylename="center-vertically"
    >
      <span
        data-stylename="text-input-label"
      >
        What's the name of your group?
      </span>
      <TextInput
        name="group-name"
        noClearButton={true}
        onChange={[Function]}
        onEnter={[Function]}
        placeholder="What's the name of your group?"
        theme={
          Object {
            "inputStyle": "modal-input",
            "wrapperStyle": "center",
          }
        }
        type="text"
        value="groupName"
      />
    </div>
  </div>
  <ModalFooter
    continueText="Continue"
    showPrevious={false}
    submit={[Function]}
  />
</div>
`;
